demos/portal-embed-demo/public/js/portalog/portals-controller.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * Default port used for the portal
+ */
+const DEFAULT_PORTAL_PORT = 3001;
+
+/**
+ * Portals Controller
+ * Controlling all the portal related features
+ */
+class PortalsController {
+
+    /**
+     * Initiating paratmeter
+     * @constructor
+     * @param {HTMLElement} embedContainer - the parent element of the embedded content
+     * @param {URL} src - URL to embed in the container
+     */
+    constructor(embedContainer, src) {
+        this.embedContainer = embedContainer;
+        this.src = src;
+        this.isPortalHostListenerAdded = false;
+
+        // Event fires when coming back from TTT Archive
+        window.addEventListener('portalactivate', (evt) => {
+            // show the scroll bar when coming back
+            document.body.classList.remove('hide-scroll-bars');
+            // embed predecessor
+            this.returnFromEmbed(evt.adoptPredecessor());
+        });
+
+        // Event fires after the portal on-click animation finishes
+        this.embedContainer.addEventListener('transitionend', (evt) => {
+
+            // We wait until the top transition finishes
+            if (evt.propertyName !== 'top') {
+                return;
+            }
+
+            const isFollowed = document.querySelector('#follow')
```

Comment:
nit: suggest moving the rest of this function into a function on PortalController (maybe `activateAfterAnimation` or something like that), probably placed after `activateAndAnimate`, to both make it clear that it's what follows that animation and to make this constructor a little more focused on hooking up the listeners.

Then the members end up following the usual order of events:
* construct
* populate embed container
* animate
* actually activate

Reply:
Fixed it!